src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -415,4 +418,20 @@ default boolean damage(double damage, DamageSource damageSource) {
         return getWorld().getEntities(predicate::test);
     }
 
+    /**
+     * Gets the current display name for this {@link Entity}.
+     *
+     * @return The display name
+     */
+    Text getDisplayName();
+
+    /**
+     * Gets the current display name for this {@link Entity}, as
+     * seen by the provided viewer.
+     *
+     * @param viewer The viewer to consider when fetching the name
+     * @return The display name
+     */
+    Text getDisplayName(@Nullable ContextViewer viewer);
```

Comment:
All the methods I've added that accept `ContextViewer` also accept `null`, which will return the value as seen without a `ContextViewer`.

In my opinion, it's cleaner to do:
```java
entity.getDisplayName(nullableViewer);
```
over
```java
if (nullableViewer == null) {
    entity.getDisplayName();
} else {
    entity.getDisplayName(nullableViewer);
}
```
everywhere.